src/storage-controller/src/collection_mgmt.rs

```diff
@@ -97,6 +110,16 @@ pub struct CollectionManager<T>
 where
     T: Timestamp + Lattice + Codec64 + TimestampManipulation,
 {
+    /// When a [`CollectionManager`] is in read-only mode it must not affect any
+    /// changes to external state.
+    ///
+    /// This is a watch to making sharing this bit with write tasks easier and
+    /// to allow write tasks to await changes.
+    pub read_only_rx: watch::Receiver<bool>,
+
+    /// Send-side for read-only bit.
+    pub read_only_tx: Arc<watch::Sender<bool>>,
```

Comment:
`CollectionManager` and the write tasks can actually flip back-and-forth. For appends, we would start returning errors again and for differential collections we'd stop writing. But I'll add an assert below (where you commented) that we only flip one way. I'll leave this as a watch because it's just easiest to implement the one-sender-multiple-receiver pattern.

Reply:
If we want the `CollectionManager` to be able to flip back to read-only mode (which seems fine!) we should probably not add an assert and instead arrange for the path to only be executed when the flag flips from `true` to `false`?

Reply:
> and instead arrange for the path to only be executed when the flag flips from true to false?

sorry, I'm not following that one 😅 

But, in general, I don't need it to be able to flip back and forth, it just happens that the easiest way (for me) to write it happened to support that.

Reply:
Sorry, I was thinking of something like this:
```rust
            _it_changed = self.read_only_watch.changed() => {
                self.read_only = self.read_only_watch.borrow();
                if !self.read_only {
                    // We can now write, attempt to do that right away, even if
                    // our `to_write` is empty. This way, we might learn that
                    // there is something in persist that we have to retract.
                    let  _ = self.write_to_persist(vec![]).await?;
                }
            }
```

But idk if that would really be enough to make `CollectionManager` support flipping back in all cases correctly. Having an assert is probably better.